sktime/forecasting/base/adapters/_pytorchforecasting.py

```diff
@@ -0,0 +1,459 @@
+        self._index_len = len(self._index_names)
+        # store X, y column names (probably None or not str type)
+        # The target column and the index will be renamed
+        # before being passed to the underlying model
+        # because those names could be None or non-string type.
+        if X is not None:
+            self._X_columns = X.columns.tolist()
+        # convert data to pytorch-forecasting datasets
+        training, validation = self._Xy_to_dataset(
+            _X, _y, self._dataset_params, self._max_prediction_length
+        )
+        if self.model_path is None:
+            # instantiate forecaster and trainer
+            self._forecaster, self._trainer = self._instantiate_model(training)
+            # convert dataset to dataloader
+            self._train_to_dataloader_params["train"] = True
+            self._validation_to_dataloader_params["train"] = False
+            # call the fit function of the pytorch-forecasting model
+            self._trainer.fit(
+                self._forecaster,
+                train_dataloaders=training.to_dataloader(
+                    **self._train_to_dataloader_params
+                ),
+                val_dataloaders=validation.to_dataloader(
+                    **self._validation_to_dataloader_params
+                ),
+            )
+            # load model from checkpoint
+            best_model_path = self._trainer.checkpoint_callback.best_model_path
+            self.best_model = self.algorithm_class.load_from_checkpoint(best_model_path)
+        else:
+            # load model from disk
+            self.best_model = self.algorithm_class.load_from_checkpoint(self.model_path)
+        return self
+
+    def _predict(
+        self: "_PytorchForecastingAdapter",
+        fh: typing.Optional[ForecastingHorizon],
+        X: typing.Optional[pd.DataFrame],
+        y: typing.Optional[pd.DataFrame],
+    ) -> pd.Series:
+        """Forecast time series at future horizon.
+
+        private _predict containing the core logic, called from predict
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+            self.cutoff
+
+        Parameters
+        ----------
+        fh : guaranteed to be ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+        X : sktime time series object, optional (default=None)
+            guaranteed to be of an mtype in self.get_tag("X_inner_mtype")
+            Exogeneous time series for the forecast
+        y : sktime time series object, optional (default=None)
+            Historical values of the time series that should be predicted.
+
+        Returns
+        -------
+        y_pred : sktime time series object
+            guaranteed to have a single column/variable
+            Point predictions
+        """
+        if y is None:
```

Comment:
The lines 236 to 241 looks like that they would be needed in all global forecasters. So I am wondering, if we should move them into the BaseGlobalForecaster.


@fkiraly what is your opinion on this?

Reply:
yes, I would agree

Reply:
although, is this the correct logic in case `y` is not `None` and `X` is?

Imagining the case where the examples are mixed exogeneous/endogeneous only.